### PR TITLE
WIP - added kms caching

### LINKED ISSUE #29: 3.6 seeing a nil:NilClass on delete_suffix
```  ×  pci-dss-3.2.1-3.6.4: [PCI-DSS 3.2.1][3.6.4] Cryptographic key changes for keys that have reached the end of their cryptoperiod.
     ×  Control Source Code Error ./controls/3.6.rb:32
     undefined method `delete_suffix' for nil:NilClass
```

### PR DESCRIPTION
**Pull request by KonradSchieban**

### CODE REVIEW

**Review by binamov — APPROVED** [review]

**Comment by KonradSchieban** [comment]
fixes https://github.com/GoogleCloudPlatform/inspec-gcp-pci-profile/issues/29
/gcbrun

File: controls/3.6.rb [hunk 1]
```diff
@@ -53,14 +53,21 @@
...
-          it "should have a lower or equal rotation period than #{kms_rotation_period_seconds}" do
-            expect(rotation_period_int).to be <= kms_rotation_period_seconds
+        if key.rotation_period.nil?
+          describe "[#{gcp_project_id}] #{key.crypto_key_name}" do
+            subject { key }
+            its('rotation_period') { should_not be_nil }
+          end
+        else
+          rotation_period_int = key.rotation_period.delete_suffix('s').to_i
+          next unless key.primary_state == "ENABLED"
```
Comment (binamov) [inline controls/3.6.rb]: we don't need to check Disabled keys at all, this probably belongs at the top of the loop

**Review by binamov — APPROVED** [review]
:shipit:  🍨 🍰

File: controls/3.6.rb (hunk 1 above)
Comment (KonradSchieban) [reply controls/3.6.rb]: Agreed and fixed. :shipit:

**Pull request merged**